src/Language/PureScript/TypeChecker/TypeSearch.hs

```diff
@@ -85,6 +110,9 @@ typeSearch
   -- ^ The typechecker state
   -> P.Type
   -- ^ The type we are looking for
-  -> Map (P.Qualified P.Ident) P.Type
+  -> ([(P.Qualified P.Ident, P.Type)], Maybe [(Label, Type)])
```

Comment:
If we do return two things, maybe it could be in a record (i.e. move the record type from `Types.hs` into this module`)?